### PR TITLE
[cleanup] addon-database: use column name instead of field index

### PR DESCRIPTION
**Pull request by howie-f**
## Description
Changes database field access from index position to column name in `AddonDatabase`

Example: `m_pDS->fv(0)...` changes to `m_pDS->fv("addonID")...`

we have no hurry with merging this as it actually doesn't fix any reported issues with the addon-system.
prevents further issues when table changes become necessary and we'll use column-name only from now on

## How Has This Been Tested?
applied permanently on live-setup and not seen regressions so far

## Types of change

- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

### CODE REVIEW

File: xbmc/addons/AddonDatabase.cpp [hunk 1]
```diff
@@ -481,7 +481,7 @@ bool CAddonDatabase::GetAddon(const std::string& addonID, const AddonVersion& ve
     if (m_pDS->eof())
       return false;
 
-    return GetAddon(m_pDS->fv(0).get_asInt(), addon);
+    return GetAddon(m_pDS->fv("id").get_asInt(), addon);
```
Comment (phunkyfish) [inline xbmc/addons/AddonDatabase.cpp:484]: Does this need to be `addons.id` or is `id` ok?

I don't actually know, I'm not much for SQL ;)

Comment (howie-f) [reply xbmc/addons/AddonDatabase.cpp:484]: if the field is not aliased -> `id` is enough, this is what is returned by the db.
if you `SELECT addons.version, repo.addonID AS repoID FROM addons` for example you need to pull **repoID**

Comment (phunkyfish) [reply xbmc/addons/AddonDatabase.cpp:484]: Cool, thanks for explaining 😉

**Review by phunkyfish — APPROVED** [review]
👍

**Review by DaveTBlake — APPROVED** [review]
Good making the db access more robust in the way

**Comment by phunkyfish** [comment]
Thanks for the contribution @howie-f

**Pull request merged**